File: Module-1/1. Array/Array Practice/2.FindSmallest/smallest.py
# smallest element in an array

def find_smallest_element(arr):
    if len(arr) == 0:
        return None  # Return None for an empty array
    smallest = arr[0]  # Initialize smallest with the first element

    for i in arr:
        if i < smallest:
            smallest = i  # Update smallest if current element is smaller
    return smallest


# Example usage:
arr = [3, 5, 7, 2, 8]
print(find_smallest_element(arr))  # Output: 2

File: Module-2-Basic-Data-Structure/1. Array/Array Practice/2.FindSmallest/smallest.js
//  Find smallest Element in an Array


const findSmallestElement = (arr) => {
  if (arr.length === 0) {
    return null; // Return null for an empty array
  }

  let smallest = arr[0]; // Initialize smallest with the first element

  for(let i = 0; i< arr.length; i++) {
    if(arr[i] < smallest) {
      smallest = arr[i]; // Update smallest if current element is smaller
    }

  }
  return smallest; // Return the smallest element
}

const arr = [5, 2, 9, 1, 5, 6];
console.log(findSmallestElement(arr)); // Output: 1